### PR TITLE
create ISSInPlace.digest test case for cIRI as reference

### PR DESCRIPTION
**Pull request by ed2k**
# Description

Add a unit test for ISSInPlace.digest. This function is used in bundle validation. It also serves the purpose of reference test implementation for cIRI.

## Type of change

- Enhancement (a non-breaking change which adds functionality)

# Checklist:

- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/test/java/com/iota/iri/crypto/ISSTest.java [hunk 1]
```diff
@@ -78,6 +78,28 @@ public void addressGenerationISS() throws Exception {
         }
     }
 
+    @Test
+    public void sigDigestISSInPlace() throws Exception {
+        int index = 0;
+        int nof = 1;
+        SpongeFactory.Mode[] modes = {SpongeFactory.Mode.KERL};
+        Hash[] hashes = {HashFactory.ADDRESS.create("DQMAVYJTIIQUCDHRIZAPYKFCFBTHNDAMHLYEEIEBSR9REQBBVUQOSDWFSGPTZFNQQTVMQPVUDHDTEZVSW")};
```
Comment (GalRogozinski) [inline src/test/java/com/iota/iri/crypto/ISSTest.java]: Can you split it to two lines after the `=`?
Our static code analysis will pick this up later on

File: src/test/java/com/iota/iri/crypto/ISSTest.java [hunk 2]
```diff
@@ -78,6 +78,28 @@ public void addressGenerationISS() throws Exception {
...
+            Converter.trits(seed, seedTrits, 0);
+
+            byte[] subseed = ISS.subseed(mode, seedTrits, index);
+            byte[] key = ISS.key(mode, subseed, nof);
+            byte[] digest = new byte[Curl.HASH_LENGTH];
+            byte[] sig = Arrays.copyOfRange(key, 0, key.length);
+            ISSInPlace.digest(mode, sig, 0, key, 0, digest);
+            byte[] address = ISS.address(mode, digest);
+            Hash addressTrytes = HashFactory.ADDRESS.create(address);
+            assertEquals(hashes[i].toString(), addressTrytes.toString());
```
Comment (GalRogozinski) [inline src/test/java/com/iota/iri/crypto/ISSTest.java]: 1. Can you please add a message in the first parameter? i.e.  "Address is not as expected"
2. The `toString()` is extraneous, `equals()` is defined :-)

File: src/test/java/com/iota/iri/crypto/ISSTest.java (hunk 1 above)
Comment (ed2k) [reply src/test/java/com/iota/iri/crypto/ISSTest.java]: not sure I understand the reason, the static code analysis will think it is too wide?

File: src/test/java/com/iota/iri/crypto/ISSTest.java (hunk 2 above)
Comment (ed2k) [reply src/test/java/com/iota/iri/crypto/ISSTest.java]: which function's first parameter? 
will remove toString(), again, the existing function also used toString(), should be removed for consistency.

Comment (kwek20) [reply src/test/java/com/iota/iri/crypto/ISSTest.java]: @ed2k every `assert()` function accepts a first parameter with a message to display when the test fails. Thats the one @GalRogozinski meant.

**Pull request merged**